src/azure-cli/azure/cli/command_modules/monitor/operations/diagnostics_settings.py

```diff
@@ -149,6 +149,8 @@ def pre_operations(self):
         export_to_resource_specific = args.export_to_resource_specific.to_serialized_data()
         if has_value(export_to_resource_specific) and export_to_resource_specific:
             args.log_analytics_destination_type = 'Dedicated'
+            if not has_value(workspace):
+                raise CLIError('usage error: --workspace and --export-to-specific-resource')
```

Comment:
CLIError is not used anymore. Please use errors defined in https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/azclierror.py